### ISSUE TITLE
Remove remaining Python 2 code

### ISSUE THREAD

**Issue by DavidFair** [mantidproject/mantid#29533]
Part of #27722 - see umbrella for context

`python3 -m compileall -f -q -j0 . > failed.txt`

The list of files that failed compileall is:

```
*** Error compiling './Framework/DataObjects/src/generate_mdevent_declarations.py'...
  File "./Framework/DataObjects/src/generate_mdevent_declarations.py", line 123
    raise IOError," can not find the string which defines the number of dimensions to process "
                 ^
SyntaxError: invalid syntax
```
```
*** Error compiling './Framework/MPIAlgorithms/scripts/CNCS_mpi_example.py'...
  File "./Framework/MPIAlgorithms/scripts/CNCS_mpi_example.py", line 21
    print "Running on rank %d of %d" % (rank, size)
                                   ^
SyntaxError: invalid syntax
```
```
*** Error compiling './Framework/MPIAlgorithms/scripts/NOM1.py'...
  File "./Framework/MPIAlgorithms/scripts/NOM1.py", line 26
    print str(size)," mpi tasks"
            ^
SyntaxError: invalid syntax
```
```
*** Error compiling './Framework/MPIAlgorithms/scripts/NOMAD_mpi_example.py'...
  File "./Framework/MPIAlgorithms/scripts/NOMAD_mpi_example.py", line 28
    print "This script must be run with fewer than 99 MPI processes! (99 banks in NOMAD)"
                                                                                        ^
SyntaxError: invalid syntax
```
```
*** Error compiling './Framework/MPIAlgorithms/scripts/PG3_mpi_example.py'...
  File "./Framework/MPIAlgorithms/scripts/PG3_mpi_example.py", line 39
    print "This script must be run with " + str(len(runs)+1) + " MPI processes!"
                                        ^
SyntaxError: invalid syntax
```
```
*** Error compiling './Framework/MPIAlgorithms/scripts/PG3_mpi_reduction.py'...
  File "./Framework/MPIAlgorithms/scripts/PG3_mpi_reduction.py", line 25
    print done
             ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print(done)?
```
```
*** Error compiling './MantidPlot/package_python_macports.py'...
  File "./MantidPlot/package_python_macports.py", line 41
    print 'Directory not copied. Error: %s' % error
                                          ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print('Directory not copied. Error: %s' % error)?
```
```
*** Error compiling './Testing/PerformanceTests/analysis.py'...
  File "./Testing/PerformanceTests/analysis.py", line 471
    print "Plotting", name
                   ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print("Plotting", name)?
```
```
*** Error compiling './Testing/PerformanceTests/check_performance.py'...
  File "./Testing/PerformanceTests/check_performance.py", line 39
    print "=============== Checking For Performance Loss ====================="
                                                                              ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print("=============== Checking For Performance Loss =====================")?
```
```
*** Error compiling './Testing/PerformanceTests/make_report.py'...
  File "./Testing/PerformanceTests/make_report.py", line 33
    print "Reading", dbfile
                  ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print("Reading", dbfile)?
```
```
*** Error compiling './Testing/PerformanceTests/reporters.py'...
  File "./Testing/PerformanceTests/reporters.py", line 44
    print '*' * nstars
            ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print('*' * nstars)?
```
```
*** Error compiling './Testing/PerformanceTests/xunit_to_sql.py'...
  File "./Testing/PerformanceTests/xunit_to_sql.py", line 77
    print "Reading", filename
                  ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print("Reading", filename)?
```
```
*** Error compiling './Testing/SystemTests/lib/systemtests/algorithm_decorator.py'...
  File "./Testing/SystemTests/lib/systemtests/algorithm_decorator.py", line 36
    print "Algorithm Parameters:"
                                ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print("Algorithm Parameters:")?
```
```
*** Error compiling './Testing/SystemTests/scripts/performance/make_report.py'...
  File "./Testing/SystemTests/scripts/performance/make_report.py", line 34
    print "Reading", dbfile
                  ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print("Reading", dbfile)?
```
```
*** Error compiling './Testing/SystemTests/scripts/performance/reporters.py'...
  File "./Testing/SystemTests/scripts/performance/reporters.py", line 44
    print '*' * nstars
            ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print('*' * nstars)?
```
```
*** Error compiling './Testing/SystemTests/scripts/performance/xunit_to_sql.py'...
  File "./Testing/SystemTests/scripts/performance/xunit_to_sql.py", line 86
    print "Reading", filename
                  ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print("Reading", filename)?
```
```
*** Error compiling './Testing/Tools/cxxtest/build_tools/SCons/cxxtest.py'...
  File "./Testing/Tools/cxxtest/build_tools/SCons/cxxtest.py", line 242
    print "Loading CxxTest tool..."
                                  ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print("Loading CxxTest tool...")?
```
```
*** Error compiling './Testing/Tools/cxxtest/python/ez_setup.py'...
  File "./Testing/Tools/cxxtest/python/ez_setup.py", line 91
    except pkg_resources.VersionConflict, e:
                                        ^
SyntaxError: invalid syntax
```
```
*** Error compiling './Testing/Tools/generatetestmain.py'...
  File "./Testing/Tools/generatetestmain.py", line 71
    print "Creating file '%s'" % filename
                             ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print("Creating file '%s'" % filename)?
```
```
*** Error compiling './buildconfig/CMake/CoverallsGenerateGcov.py'...
  File "./buildconfig/CMake/CoverallsGenerateGcov.py", line 86
    print "YES: ",sourceWithPath.strip()," WAS FOUND"
                ^
SyntaxError: invalid syntax
```
```
*** Error compiling './buildconfig/Jenkins/oclint.py'...
  File "./buildconfig/Jenkins/oclint.py", line 50
    print 'linting ... %s' %file_name
                         ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print('linting ... %s' %file_name)?
```
```
*** Error compiling './buildconfig/doxygen_to_sip.py'...
  File "./buildconfig/doxygen_to_sip.py", line 47
    print "WARNING: Could not find method %s" % method
                                            ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print("WARNING: Could not find method %s" % method)?
```
```
*** Error compiling './buildconfig/python_export_maker.py'...
  File "./buildconfig/python_export_maker.py", line 92
    print 'Writing export file \"%s\" '% os.path.basename(exportfile)
                                      ^
SyntaxError: invalid syntax
```
```
*** Error compiling './buildconfig/sip_strip_docstring.py'...
  File "./buildconfig/sip_strip_docstring.py", line 89
    print "---- Stripping docstrings from %s ---- " % options.sipfile
                                                  ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print("---- Stripping docstrings from %s ---- " % options.sipfile)?
```
```
*** Error compiling './qt/scientific_interfaces/CreateInterfaceTemplate.py'...
  File "./qt/scientific_interfaces/CreateInterfaceTemplate.py", line 25
    print "Usage: CreateInterfaceTemplate interface_name yes|no\n" \
                                                                 ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print("Usage: CreateInterfaceTemplate interface_name yes|no\n" \)?
```
```
*** Error compiling './qt/widgets/plugins/algorithm_dialogs/customDialogTemplate.py'...
  File "./qt/widgets/plugins/algorithm_dialogs/customDialogTemplate.py", line 23
    print "Usage: createDialogTemplate algorithm-name yes|no\n" \
                                                              ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print("Usage: createDialogTemplate algorithm-name yes|no\n" \)?
```
```
*** Error compiling './qt/widgets/sliceviewer/doc/sliceviewer_wiki_screenshots.py'...
  File "./qt/widgets/sliceviewer/doc/sliceviewer_wiki_screenshots.py", line 64
    print dir(sv)
            ^
SyntaxError: invalid syntax
```

**Comment by joseph-torsney** [comment]
`./qt/widgets/sliceviewer/doc/sliceviewer_wiki_screenshots.py` and `./MantidPlot/package_python_macports.py` have since been removed

**Comment by joseph-torsney** [comment]
This issue is about to be closed via #30819, but not all of the python2 code has been removed. Other than the MPI scripts, I'm quite sure that we don't want to remove the other files, so we should consider migrating the following to python3:
- [Framework/DataObjects/src/generate_mdevent_declarations.py](https://github.com/mantidproject/mantid/blob/23fb6a7963e1688ca77be72a61716c42c58e4c73/Framework/DataObjects/src/generate_mdevent_declarations.py)
- [Performance Tests](https://github.com/mantidproject/mantid/tree/master/Testing/PerformanceTests)
- [Testing/SystemTests/lib/systemtests/algorithm_decorator.py](https://github.com/mantidproject/mantid/blob/23fb6a7963e1688ca77be72a61716c42c58e4c73/Testing/SystemTests/lib/systemtests/algorithm_decorator.py)
- [Testing/Tools/generatetestmain.py](https://github.com/mantidproject/mantid/blob/23fb6a7963e1688ca77be72a61716c42c58e4c73/Testing/Tools/generatetestmain.py)
- [qt/scientific_interfaces/CreateInterfaceTemplate.py](https://github.com/mantidproject/mantid/blob/23fb6a7963e1688ca77be72a61716c42c58e4c73/qt/scientific_interfaces/CreateInterfaceTemplate.py)
- (qt/widgets/plugins/algorithm_dialogs/customDialogTemplate.py)[https://github.com/mantidproject/mantid/blob/23fb6a7963e1688ca77be72a61716c42c58e4c73/qt/widgets/plugins/algorithm_dialogs/customDialogTemplate.py]

The rest of the files, such as those in `buildconfig/` and `Testing/toold/cxxtest` are not ours, so we should leave them alone.
